src/Microsoft.AspNet.OData.Shared/IDeltaSetItem.cs

```diff
@@ -0,0 +1,38 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="IDeltaSetItem.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Basic interface for representing a delta item like delta, deleted entity, etc.
+    /// </summary>
+    public interface IDeltaSetItem
+    {
+        /// <summary>
+        /// Gets the kind of object within the delta payload.
+        /// </summary>
+        EdmDeltaEntityKind DeltaKind { get; }
+
+        /// <summary>
+        /// Gets or sets the annotation container to hold transient instance annotations.
+        /// </summary>
+        IODataInstanceAnnotationContainer TransientInstanceAnnotationContainer { get; set; }
```

Comment:
Why don't we have the `PersistentInstanceAnnotationContainer` here?

Reply:
I tend to believe an `IDeltaSetItem` does not hold `persistent annotations.`